Form/Type/DocumentType.php

```diff
@@ -20,9 +20,24 @@
 use Doctrine\Bundle\PHPCRBundle\Form\ChoiceList\PhpcrOdmQueryBuilderLoader;
 use Symfony\Bridge\Doctrine\Form\ChoiceList\EntityLoaderInterface;
 use Symfony\Bridge\Doctrine\Form\Type\DoctrineType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
 
 class DocumentType extends DoctrineType
 {
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        parent::setDefaultOptions($resolver);
+        $type = $this;
+        $loader = function (Options $options) use ($type) {
+            $qb = $options['query_builder'] ?: $options['em']->createQueryBuilder()->fromDocument($options['class'], 'a');
+
+            return $type->getLoader($options['em'], $qb, $options['class']);
```

Comment:
imho increasing the min version or setting conflicts are the way to go. less code to maintain for us.